core/src/execution/datafusion/expressions/cast.rs

```diff
@@ -107,7 +108,23 @@ macro_rules! cast_utf8_to_timestamp {
         result
     }};
 }
-
+macro_rules! cast_utf8_to_date {
```

Comment:
Any reason why this is a macro and not a function? I see only one usage (maybe I missed something).

Reply:
Hi @parthchandra thanks for the review. I am new to rust (and also my first attempt at an oss contribution). 
i was imitating the code from cast_utf8_to_timestamp. removed it now.

Reply:
> I am new to rust

So am I :). Welcome to the community !